server/config/swagger.js

```diff
@@ -0,0 +1,27 @@
+import swaggerJsdoc from 'swagger-jsdoc';
+import { config } from 'dotenv';
+
+// Initialize dotenv
+config();
+
+// Swagger Definitions
+const swaggerDefinition = {
+  info: {
+    title: 'Author\'s Haven',
+    version: '1.0.0',
+    description: 'A Social platform for the creative at heart',
+  },
+  host: `${process.env.HOST_NAME}` || 'localhost:3000',
```

Comment:
placing this in a string like this This causes a problem if `process.env.HOST_NAME` does not exist we have a string `'undefined'`  which is a valid string.

I fail to understand why you have this like this despite my telling you otherwise